lib/galaxy/workflow/run.py

```diff
@@ -231,12 +231,17 @@ def remaining_steps(self):
         remaining_steps = []
         step_invocations_by_id = self.workflow_invocation.step_invocations_by_step_id()
         for step in steps:
+            step_id = step.id
             if not hasattr( step, 'module' ):
                 self.module_injector.inject( step )
-                runtime_state = step_states[ step.id ].value
+                if step_id not in step_states:
+                    template = "Workflow invocation [%s] has no step state for step id [%s]. States are [%s]."
+                    message = template % (self.workflow_invocation.id, step_id, step_states)
```

Comment:
step_states is a iteratable and this will result in an output similar to `[[1,2,3]]` not sure this is intended.